ya-erigon-runtime/src/main.rs

```diff
@@ -0,0 +1,208 @@
+#[conf(ErigonConf)]
+pub struct ErigonRuntime {
+    seq: AtomicU64,
+    erigon_pid: Option<Child>,
+    rpcdaemon_pid: Option<Child>,
+    host: Option<String>,
+}
+
+impl Runtime for ErigonRuntime {
+    const MODE: RuntimeMode = RuntimeMode::Server;
+
+    fn deploy<'a>(&mut self, ctx: &mut Context<Self>) -> OutputResponse<'a> {
+        let data_dir_path = PathBuf::from(
+            ctx.conf
+                .data_dir
+                .clone()
+                .expect("Configuration data_dir is missing"),
+        );
+        let parent_data_dir = data_dir_path.as_path();
+        if !(parent_data_dir.exists() && parent_data_dir.is_dir()) {
+            panic!(
+                "Configuration 'data_dir' directory has to exists and needs to contain
+                 the directory for every supported network, named of the network"
+            );
+        }
+
+        async move {
+            Ok(serialize::json::json!(
+                {
+                    "startMode":"blocking",
+                    "valid":{"Ok":""},
+                    "vols":[]
+                }
+            ))
+        }
+        .boxed_local()
+    }
+
+    fn start<'a>(&mut self, ctx: &mut Context<Self>) -> OutputResponse<'a> {
+        //TODO: Receive chain_id from start parameters when Yapapi ready
+        // see: https://github.com/golemfactory/yapapi/issues/390
+        let chain_id = String::from("goerli");
+        let current_exe_path = std::env::current_exe().unwrap();
+        let path = current_exe_path.parent().unwrap();
+
+        let data_dir_path = prepare_data_dir_path(
+            &ctx.conf
+                .data_dir
+                .clone()
+                .expect("Configuration data_dir is missing"),
+            &chain_id,
+        );
+        let mut erigon_pid = spawn_process(
+            &mut Command::new(&path.join(ERIGON_BIN))
+                .arg("--chain")
+                .arg(chain_id),
+            &data_dir_path,
+            &[""; 0],
+        )
+        .expect("Erigon: Failed to spawn");
+
+        let rpcd_pid = spawn_process(
+            &mut Command::new(&path.join(RPCDAEMON_BIN)),
+            &data_dir_path,
+            RPCDAEMON_PARAMS,
+        );
+        if !rpcd_pid.is_ok() {
+            let _ = (&mut erigon_pid).kill();
+        }
+        let rpcd_pid = rpcd_pid.expect("RPC Daemon: Failed to spawn");
+
+        self.erigon_pid = Some(erigon_pid);
+        self.rpcdaemon_pid = Some(rpcd_pid);
+
+        self.host = ctx.conf.public_addr.clone();
+        async move { Ok("start".into()) }.boxed_local()
+    }
+
+    fn stop<'a>(&mut self, _: &mut Context<Self>) -> EmptyResponse<'a> {
+        let mut erigon_kill_ok = false;
+        let mut rpcd_kill_ok = false;
+        if let Some(erigon_pid) = &mut self.erigon_pid {
+            erigon_kill_ok = erigon_pid.kill().is_ok();
+        }
+        if let Some(rpcd_pid) = &mut self.rpcdaemon_pid {
+            rpcd_kill_ok = rpcd_pid.kill().is_ok();
+        }
```

Comment:
Mutable variables are lame. :stuck_out_tongue: 
```suggestion
        let erigon_kill_ok = match &mut self.erigon_pid {
            Some(erigon_pid) => erigon_pid.kill().is_ok(),
            None => false,
        };
        let rpcd_kill_ok = match &mut self.rpcdaemon_pid {
            Some(rpcdaemon_pid) => rpcdaemon_pid.kill().is_ok(),
            None => false,
        };
```